### ISSUE TITLE
Improving seq2seq chitchat model OR pretrained model

### ISSUE THREAD

**Issue by gmkim90** [facebookresearch/ParlAI#283]
Hello.
I am currently training seq2seq model on Opensubtitle dataset.

Is there anyone provide advice improving training of seq2seq?
or 
Is there anyone provide pretrained model as an example?

My training detail is as follows :

python examples/train_model.py -m seq2seq -t opensubtitles -mf data/OpenSubtitles/model_512 -hs 512 -bs 64  --gpu 0 -lr 0.001
python examples/train_model.py -m seq2seq -t opensubtitles -mf data/OpenSubtitles/model_2048 -hs 2048 -bs 64 --gpu 1 -lr 0.001

I am expecting similar quality with references.
[1] O. Vinyals et al., "A Neural Conversational Model", 2015 (https://arxiv.org/abs/1506.05869)
[2] J. Li et el., "A Diversity-Promoting Objective Function for Neural Conversation Model", 2016 (https://arxiv.org/abs/1510.03055)

Training loss seems not decreasing significantly, but prediction & test result is non-sense.
Specifically,
(1) It sometimes output same response.
Enter Your Message: hi
[Seq2Seq]: - No .
Enter Your Message: hello
[Seq2Seq]: - No .
Enter Your Message: how are you?
[Seq2Seq]: - No .
Enter Your Message: do you know me?
[Seq2Seq]: - No .

(2) It sometimes output non-sentence.
Enter Your Message: hi
[Seq2Seq]: Mayan Tryin whinny Marcelo , Elisheba Mayan trough dependin ForBroke
Enter Your Message: hello
[Seq2Seq]: Collateral ,
Enter Your Message: how are you?
[Seq2Seq]: That\ ' s right ,
Enter Your Message: do you know me?
[Seq2Seq]: That\ ' s right ,

**Comment by alexholdenmiller** [comment]
Hi @gmkim90, thanks for flagging! I'll be pushing a series of improvements to the model over the next week, hopefully they'll help to resolve this.

**Comment by jsamson008** [comment]
I'm just curious if the improvements to the model have been pushed.

**Comment by alexholdenmiller** [comment]
@jsamson008 yes! The model is now separated into a subclass of nn.Module so it should be easier to customize and work with, and I just pushed performance improvement to other parts of parlai as well.

Note that we recommend using the `-tr` setting when training to truncate examples and speed up training--we don't support batching by example size currently, so you can get a lot of padding from unusually long examples, but this is mitigated by truncation.

**Comment by Ravi-Jay** [comment]
@alexholdenmiller Can you show us how to use the pre-train model, I use tensorflow,mxnet ,both get bad result, really thx!

**Comment by jsamson008** [comment]
I second the chatbot checkpoint to reproduce results. 

@Ravi-Jay - I think that the neural conversation model uses the OpenSubtitles 2013 dataset. This could explain why it is difficult to get the desired results. Also, which implementation of J. Li et al. are you using? From my experience there are incorrect implementations. Make sure you use https://github.com/jiweil/Neural-Dialogue-Generation. If you have done this withing parlai, can you submit a patch?

**Comment by jsedoc** [comment]
For standard Seq2Seq with OpenNMT-py checkpoints you can go [here](https://github.com/jsedoc/SETC/blob/master/README.md). In our paper, we found that according to our human evaluation these checkpoints are worse than the original NCM paper. However, [Cakechat](https://github.com/lukalabs/cakechat) was quite a bit closer and I recommend that you check it out. Cakechat checkpoints are freely available.

**Comment by jsedoc** [comment]
@Ravi-Jay @alexholdenmiller @gmkim90 OpenNMT also has a pre-trained model for opensubtitles available http://opennmt.net/Models-py/.

**Comment by stephenroller** [comment]
Closing due to age. The convai2 baseline models are a good place to start, and the recent BERT efforts (#1331).
